ComplexDateTimeField should fallback to None if null=True

jimmyshen opened the pull request:
Hi,

I was playing with the new 'null' behavior, which I would like to use in a project I'm working on. While messing with it on master, I noticed some inconsistency in the behavior with regard to DateTimeField and ComplexDateTimeField.

Here's a demonstration:

```python
def test_null_datetime_fields():
    class TestDoc(Document):
        dt_field = DateTimeField(null=True)
        cdt_field = ComplexDateTimeField(null=True)

    doc = TestDoc()
    doc.save()

    assert doc.dt_field is None
    assert doc.cdt_field is None
```

The cause seems to be related to some logic specific to ComplexDateTimeField that falls back to datetime.now on retrieval from db. I'm not sure what the motivation was for the different behavior between DateTimeField and ComplexDateTimeField -- it was introduced a long time ago and there's even a test to ensure it doesn't regress:

https://github.com/MongoEngine/mongoengine/commit/0187a0e1

Thoughts?

DavidBord commented:
Could you please add your test to your commit and yourself to the AUTHORS file?

jimmyshen commented:
Done! I extended my test to cover a few new fields too since it didn't look like there were any tests for null=True yet

jimmyshen commented:
Whoops! That was a silly mistake. Pushed the suggested updates and refreshed from upstream.

DavidBord commented:
It will be nicer if you would have added the extra fields to the existing class in the test instead of duplicating some of them to another class
In addition, please squash your commits

jimmyshen commented:
Sorry, I was busy for a while. I updated the test per your comment; squashed my changes, had to force push them over the existing branch and then I re-merged upstream master.

DavidBord commented:
Thanks
One last thing I forgot - please add the change to `changelog.rst`

jimmyshen commented:
Done
